libs/api.lua

```diff
@@ -130,6 +131,44 @@ end
 
 local metaCache = {}
 
+-- Collect statistics about this server's running instance.
+-- Return these metrics in a table.  The intended consumer of this
+-- information should render it in JSON format for transmission to
+-- a time-series database and/or visualization tools.
+local function collectStats()
+  local entry, fdsUsed = 0, 0
+  local memoryUsed = 1024 * collectgarbage("count")
+
+  local function countFDs(path)
+    local entries = 0
+
+    -- You might think this leaks a filehandle, but it actually doesn't.
+    -- You can confirm this yourself by hitting /metrics several times in a row,
+    -- and observing that lua.fds.total does not increase with each GET request.
+
+    for entry in io.popen([[ls -1 "]]..path..[["]]):lines() do
```

Comment:
Here is a better example https://github.com/luvit/lit/blob/b3590f4bf5d4e271126f8ed2b6f66462c9fbe21d/libs/get-installed.lua#L25-L27